.github/pull_request_template.md

```diff
@@ -16,13 +16,16 @@
     of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
     description heading and to add a 'x' to the correct box.
 -->
-    [ ] Fix
-    [ ] Refactor
-    [ ] Target update
-    [ ] Functionality change
-    [ ] Docs update
-    [ ] Test update
-    [ ] Breaking change
+    [ ] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
+    [ ] Feature update (New feature / Functionality change / New API)
```

Comment:
If a PR is a feature  update, can we add a sub-question to ask them submitter to confirm where is the documentation and tests related to this new feature. @adbridge

Reply:
as if "new API or feature, documentation and tests provided are required" ?

Reply:
@jamesbeyond We already have a rest results (optional) section there. Are you saying we should expand on that to also show evidence for new feature testing ? I'm happy to do that. We could also have a section for 'Documentation Link'. What do @0xc0170 and @bulislaw think?

Reply:
`Test Result` would make people think the CI that running would provide the test result.
The really question here is: if your PR is about a new feature, where is your tests ? are these test can be executed by current CI ?
Maybe we can make the section compulsory, but let them choose one from bellow?
[ ] No Tests required
[ ] This PR changes will be covered by current CI
[ ] Tests supplied in this PR
[ ] Other, specify bellow

BTW, I'd like to have the Documentation link section as well

Reply:
Mmm the only problem I have with this is we are trying to simplify the basic PR template and this would start to add complexity again. Though I do like the idea @bulislaw @0xc0170 what are your thoughts ?

Reply:
I think our goal is trying formalize and clarify the process, make the guidance easy for users to follow, and make it easy for us to manage the PRs. Not just only simplify it 😉

Reply:
The problem with above is for many people CI tests are unkown and even testing infrastructure, they test it via their app or simple test case, so this would be mostly for internal teams rather than all of us. 

What we often do is ask a simple question "How was this tested", maybe this type of question to be included in the template - it should make a person think about testing this. The answer could be one of the above "no tests done" , "tested on simple app", "run entire test suite", etc. I still feel we don't test enough would this question asked for all PRs help us to improve test coverage?

Reply:
Yes @0xc0170 I agree with you. So the things I am a bit worried is about when people adding new feature, they more focus about the feature and forgot about tests and documentation.
e.g. the baremetal profile. Having this test compulsory section would let aware that tests are required as part of feature.

Also `tested on simple app` it is not acceptable to me for a new feature. we need them either in formal of `Greentea`, `Unittest` format or a test in their own team's CI, unless they'll need a really good explanation why there is no test.
For the changes to a existing feature, it is fairly enough for them to claim this is already covered by a exist `greentea` or `unittest` tests

Where is test is one of the question, and we know there are limitation about CI, e.g. USB test is is not running. I think that's another issue for CI/Infra team to resolve